hazelcast/src/main/java/com/hazelcast/query/impl/AbstractIndex.java

```diff
@@ -220,12 +223,16 @@ public PerIndexStats getPerIndexStats() {
     }
 
     private Object extractAttributeValue(Data key, Object value) {
+
         if (components == null) {
-            return QueryableEntry.extractAttributeValue(extractors, ss, name, key, value, null);
+            QueryEntry queryEntry = new QueryEntry(ss, key, value, extractors);
```

Comment:
Json metadata never reaches here anyway. Indexing system does not pass along the whole entry but only key and value. So we lose metadata along the way. It is not a big deal since indexing is once per put.

Anyway, passing too many arguments around is not the issue. I was trying to reduce code duplication in `MapEntryAttributeExtractor`. Passing key/value instead of query leads to having two `extractAttributeValue` methods which are very similar. I'll try to find a better way.

Reply:
Actually, `AbstractIndex.putEntry` accepts `QueryableEntry`, so in theory it should be possible to obtain JSON metadata at least for this passed entry. But yes, it's not possible for other entries that are constructed from keys/values.

Reply:
BTW, is it possible to reuse the metadata from the passed entry in `AbstractIndex.putEntry` for `oldAttributeValue` extraction? If that's possible, the only problematic case is `AbstractIndex#removeEntry`.

Reply:
No, that metadata is associated with the new value. We may try to pass old metadata to AbstractIndex.putEntry too but I don't think it is worth the additional complexity. Extracting an attribute even without metadata is a small part of put overhead. Still, we can make use of metadata where possible like here in extracting `newAttributeValue`.